### ISSUE TITLE
`info USD` (or any alias of `USD`) doesn't work until a currency calculation is performed

### ISSUE THREAD

**Issue by rben01** [sharkdp/numbat#635]
```
>>> info USD
  Not found
>>> 1USD

  1 dollar

    = 1 $    [Money]

>>> info USD
  Unit: US dollar (https://en.wikipedia.org/wiki/United_States_dollar)
  Aliases: dollar, dollars, USD, usd, $
  A unit of: Money
  
  1 dollar = 0.92584 €
```

I assume that this is related to the lazy loading of currencies. Should currencies be loaded if `info` triggers a use?

**Comment by sharkdp** [comment]
Thank you very much for reporting this. This lazy loading of currencies really causes a lot of issues. We've been recently discussing a general lazy evaluation strategy that would potentially solve this alltogether.

> Should currencies be loaded if `info` triggers a use?

Yes. A similar problem exists with running `list`. I think we should probably trigger a load of the currency module whenever a command is run (assuming that we are not in no-prelude mode). Basically the same that we do here:

https://github.com/sharkdp/numbat/blob/77decee691bf7fe77eeacdf313bc894edf084468/numbat/src/lib.rs#L809-L833

**Comment by rben01** [comment]
In theory we don't have to eagerly load the currencies on the first command — if it's `info`, we can look at the argument and then use the same logic as in `Context::interpret_with_settings` to decide to load the currencies. Same goes for `list` and `list units` (but not `list functions`). But no need to load currencies for `save`, `help`, and especially `quit`.

Separately, looking at the comment in that function

> // TODO: maybe we can somehow load this list of identifiers from units::currencies?

Tackling this seems easy enough with a build script that parses `units/currencies.nbt` and emits a Rust array to a file, and then `const CURRENCY_IDENTIFIERS: &[&str] = include!(...)`.

**Comment by sharkdp** [comment]
> Tackling this seems easy enough with a build script that parses `units/currencies.nbt` and emits a Rust array to a file, and then `const CURRENCY_IDENTIFIERS: &[&str] = include!(...)`.

Yes. Let's maybe wait for the results of the *"should Numbat do lazy evaluation by default?"* discussion before investing time here though.
